source_manager.go

```diff
@@ -27,6 +27,18 @@ var sanitizer = strings.NewReplacer(":", "-", "/", "-", "+", "-")
 // sufficient for any purpose. It provides some additional semantics around the
 // methods defined here.
 type SourceManager interface {
+	// SourceManager adds a Release method to the sourceDeducer method set. The
+	// sourceDeducer method set is defined in a separate interface because it's
+	// used by the sourceBridge interface as well, but sourceBridges don't get a
+	// Release method.
+	sourceDeducer
```

Comment:
Okay, that works for me. Kind of a shame to have the methods written in two places, but if you want them to not use contexts in the bridge, it makes sense I suppose.

Kind of an aside, but gps might benefit from some renaming of some these interfaces, maybe breaking them into composable chunks a bit. I don't have anything _too_ concrete, but names like "manager" and "bridge" and "analyzer" and "deducer" are all pretty similar.

It would probably be way easier to do that https://github.com/golang/dep/issues/300 is solved.